go/libraries/doltcore/sqle/tables.go

```diff
@@ -1062,18 +1061,39 @@ func (t *AlterableDoltTable) CreateForeignKey(
 	if err != nil {
 		return err
 	}
+
+	tableIndex, ok := t.sch.Indexes().GetIndexByTags(colTags...)
+	if !ok {
+		// if child index doesn't exist, create it
+		ret, err := doltdb.CreateIndexForTable(ctx, t.table, "", sql.IndexUsing_Default, sql.IndexConstraint_None, sqlColNames, false, "")
+		if err != nil {
+			return err
+		}
+		t.table = ret.NewTable
+		tableIndex = ret.NewIndex
+		root, err = root.PutTable(ctx, t.name, t.table)
+		if err != nil {
+			return err
+		}
+	}
+
 	var refTbl *doltdb.Table
-	var ok bool
 	var refSch schema.Schema
 	if isSelfFk {
 		refTbl = t.table
 		refSch = t.sch
 	} else {
+		// TODO: Look here to see that then when the referenced table does not exist an error is thrown
 		refTbl, _, ok, err = root.GetTableInsensitive(ctx, refTblName)
 		if err != nil {
 			return err
 		}
+
+		checksDisabled := isForeignKeyChecksDisabled(ctx)
 		if !ok {
+			if checksDisabled {
+				return t.addEmptyForeignKeyToCollection(ctx, root, fkName, tableIndex.Name(), colTags, refTblName, refColumns, onUpdate, onDelete)
```

Comment:
Can't link to it as it's not edited, but we're duplicating quite a bit of work in `ForeignKey.ResolveReferencedIndexAndTags`. Instead, you can just create the foreign key here as though it's going to be an empty one every time, and if foreign key checks are _enabled_, then we immediately resolve them.